### DISCUSSION TITLE
How to calibrate using TWS

### DISCUSSION THREAD

**Discussion by weixiaoxun** [iiasa/CWatM#162] [Q&A]
Dear @dof1985 @mikhailsmilovic ,

I hope this message finds you well.

I am currently working on calibrating the model using GRACE's Terrestrial Water Storage (TWS) data, replacing the discharge observation data. However, when I run the script 1a. run_single_calibration, I encounter the following error:

Exception: No simulated streamflow found. Probably the model failed to start? Check the log files of the run!

I have replaced all instances of discharge with TWS in the script. Could this error be due to a missing or incorrect variable that needs to be changed in the script to account for the use of TWS data instead of streamflow? I would appreciate any advice on whether additional adjustments are needed to accommodate this data substitution.

Thank you for your assistance.
![微信截图_20241010204321](https://github.com/user-attachments/assets/051f2900-1910-419e-952c-779035e1950e)
[calibration_single.txt](https://github.com/user-attachments/files/17328286/calibration_single.txt)
[settings_calibration.txt](https://github.com/user-attachments/files/17328292/settings_calibration.txt)

Best regards,
Xiaoxun

**Comment by dof1985** [comment]
Hi @weixiaoxun 

Some few hints:

1/ This error indicate that your model runs have not completed as expected and guide you to check the log files (located in the catchment folder). It may be just some problems with your settings file the cause a model crash (note there is another settings file, that you haven't shared, end with .ini)
2/ Another potential cause - maybe you haven't added the tws to your outputs as tss in your settings file.

Good luck
Dor

**Reply by weixiaoxun** [reply]
Dear Professor @dof1985 ,

Thank you very much for taking the time to respond to my question. I greatly appreciate your willingness to assist me with the issue I encountered.

As requested, I have attached the ini file in text format for your reference. I would be extremely grateful if you could help me troubleshoot the problem.

Thank you again for your valuable time and support.
[settings_CWatM_template.txt](https://github.com/user-attachments/files/17388247/settings_CWatM_template.txt)

Best regards,
Weixiaoxun
